test/src/androidTest/kotlin/io/realm/NotificationTests.kt

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import io.realm.runtimeapi.RealmModule
+import io.realm.util.RunLoopThread
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import test.Sample
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import kotlin.time.ExperimentalTime
+import kotlin.time.seconds
+
+private val INITIAL = "Hello, World!"
+private val FIRST = "FIRST"
+private val SECOND = "SECOND"
+
+@OptIn(ExperimentalTime::class)
+class NotificationTests {
```

Comment:
Already made an `RunLoopThread`-abstraction, but we cannot put it in `commonTest` and sym-link as it will then create overlapping symbols. Will add a FIXME with link to the youtrack issue.